logging/proto_conversions_test.go

```diff
@@ -0,0 +1,128 @@
+package logging
+
+import (
+	"testing"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.viam.com/test"
+)
+
+func TestFieldConversion(t *testing.T) {
+	t.Parallel()
+
+	testTime, err := time.Parse(DefaultTimeFormatStr, "2024-03-01T11:39:40.897Z")
+	test.That(t, err, test.ShouldBeNil)
+
+	type s struct {
+		Field1 string
+		Field2 bool
+	}
+	testStruct := &s{"foo", true}
+
+	testCases := []struct {
+		field       zap.Field
+		expectedVal any
+	}{
+		{
+			field: zap.Field{
+				Key:     "boolean",
+				Type:    zapcore.BoolType,
+				Integer: 0,
+			},
+			expectedVal: false,
+		},
+		{
+			field: zap.Field{
+				Key:     "duration",
+				Type:    zapcore.DurationType,
+				Integer: time.Hour.Nanoseconds(),
+			},
+			expectedVal: time.Hour,
+		},
+		{
+			field: zap.Field{
+				Key:     "float64",
+				Type:    zapcore.Float64Type,
+				Integer: 4608218246714312622,
+			},
+			expectedVal: float64(1.23),
+		},
+		{
+			field: zap.Field{
+				Key:     "float32",
+				Type:    zapcore.Float32Type,
+				Integer: 1068037571,
+			},
+			expectedVal: float32(1.32),
```

Comment:
Would you mind adding a test for a large exactly representable float32? (something like 2^40 or 2^80?)

https://en.wikipedia.org/wiki/Single-precision_floating-point_format
```
All integers with 7 or fewer decimal digits, and any 2n for a whole number −149 ≤ n ≤ 127, can be converted exactly into an IEEE 754 single-precision floating-point value. 
```

I am asking because I wonder if we shouldn't also throw float32's into strings

Reply:
Yeah good idea; added tests for `MaxFloat64` and `MaxFloat32`.